# Results comparison for MIST filtering

In this notebook, I will compare the results of the different models relative to how the data was filtered (only the MIST dataset).

I will first look at how the filter on one value affected the other (if it didn't affect it, I will not write it) and then see how it affected the results of each model.

In [1]:
from ML.utils.utils import *
from physics.Iso_data_handler import Iso_data_handler
from ML.utils.Data_preparator import Data_preparator

In [2]:
pre_path = "../../../../../../"
path = pre_path + "results/model_A_partial/default_models"

In [3]:
compare_metrics(path, 
                output_parameters=["mass"], # output_parameters=["radius"],
                model_names=["random_forest"],
                physical_models=["MIST"],
                data_filters = ["Base", "mass_smaller_3", "mass_smaller_14", "mass_smaller_30"])

Results for the Global category with a value of 1.0


value range  \
model         physical_model filter          output_parameter                                            
random_forest MIST           Base            mass               0.0999979840073621 - 298.5447575808816   
                             mass_smaller_3  mass              0.0999979840073621 - 2.9999868374941867   
                             mass_smaller_14 mass               0.0999979840073621 - 298.5447575808816   
                             mass_smaller_30 mass               0.0999979840073621 - 298.5447575808816   

                                                                              RVE  \
model         physical_model filter          output_parameter                       
random_forest MIST           Base            mass              0.9978054138856324   
                             mass_smaller_3  mass              0.9992541969730512   
                             mass_smaller_14 mass              0.9978246279671578   
                             mass_smaller_30 mass              0.9978090814796585   

                                                                              RMSE  \
model         physical_model filter          output_parameter                        
random_forest MIST           Base            mass               1.2512629515795346   
                             mass_smaller_3  mass              0.01975564332021666   
                             mass_smaller_14 mass               1.2457736076457506   
                             mass_smaller_30 mass               1.2502174142307747   

                                                                                MAE  \
model         physical_model filter          output_parameter                         
random_forest MIST           Base            mass                0.1452769419896218   
                             mass_smaller_3  mass              0.006025796628853846   
                             mass_smaller_14 mass                0.1450974561041155   
                             mass_smaller_30 mass               0.14544775978054023   

                                                                               MedAE  \
model         physical_model filter          output_parameter                          
random_forest MIST           Base            mass               0.004111252117102282   
                             mass_smaller_3  mass              0.0020703142458058377   
                             mass_smaller_14 mass               0.004128173546273484   
                             mass_smaller_30 mass               0.004125628598466324   

                                                                             CORR  \
model         physical_model filter          output_parameter                       
random_forest MIST           Base            mass              0.9989035373674631   
                             mass_smaller_3  mass              0.9996270807690646   
                             mass_smaller_14 mass              0.9989131374040815   
                             mass_smaller_30 mass              0.9989052645073933   

                                                                           MAX_ER  \
model         physical_model filter          output_parameter                       
random_forest MIST           Base            mass               95.14828406903314   
                             mass_smaller_3  mass              1.3490965095666518   
                             mass_smaller_14 mass                99.1425528233774   
                             mass_smaller_30 mass              103.44266579660032   

                                                                                                               Percentiles  \
model         physical_model filter          output_parameter                                                                
random_forest MIST           Base            mass                 75 : 0.0156 / 90 : 0.0791 / 95 : 0

Results for the phase category with a value of 0.0


value range  \
model         physical_model filter          output_parameter                                            
random_forest MIST           Base            mass               0.0999979840073621 - 298.5447575808816   
                             mass_smaller_3  mass              0.0999979840073621 - 2.9999868374941867   
                             mass_smaller_14 mass               0.0999979840073621 - 298.5447575808816   
                             mass_smaller_30 mass               0.0999979840073621 - 298.5447575808816   

                                                                              RVE  \
model         physical_model filter          output_parameter                       
random_forest MIST           Base            mass              0.9973575906750899   
                             mass_smaller_3  mass               0.999947044394285   
                             mass_smaller_14 mass               0.997413901088023   
                             mass_smaller_30 mass               0.997368944119941   

                                                                               RMSE  \
model         physical_model filter          output_parameter                         
random_forest MIST           Base            mass                1.3521665845754873   
                             mass_smaller_3  mass              0.005488828615455466   
                             mass_smaller_14 mass                1.3376914087667897   
                             mass_smaller_30 mass                1.3492653913806696   

                                                                                MAE  \
model         physical_model filter          output_parameter                         
random_forest MIST           Base            mass               0.16783299547779296   
                             mass_smaller_3  mass              0.002764338682723445   
                             mass_smaller_14 mass               0.16765041145105403   
                             mass_smaller_30 mass               0.16841881212380835   

                                                                               MedAE  \
model         physical_model filter          output_parameter                          
random_forest MIST           Base            mass              0.0035490560629605206   
                             mass_smaller_3  mass              0.0013470803912696283   
                             mass_smaller_14 mass              0.0035453418124899994   
                             mass_smaller_30 mass              0.0035401103163561487   

                                                                             CORR  \
model         physical_model filter          output_parameter                       
random_forest MIST           Base            mass              0.9986793327165373   
                             mass_smaller_3  mass              0.9999735435321911   
                             mass_smaller_14 mass              0.9987075486293543   
                             mass_smaller_30 mass               0.998685112002334   

                                                                            MAX_ER  \
model         physical_model filter          output_parameter                        
random_forest MIST           Base            mass                83.61091180145692   
                             mass_smaller_3  mass              0.18157378789633682   
                             mass_smaller_14 mass                83.57250382102333   
                             mass_smaller_30 mass                82.85120835326344   

                                                                                                               Percentiles  \
model         physical_model filter          output_parameter                                                                
random_forest MIST           Base            mass                75 : 0.01627 / 90 : 0.1

Results for the phase category with a value of 2.0


value range  \
model         physical_model filter          output_parameter                                            
random_forest MIST           Base            mass              0.6538581961218565 - 283.72102985530483   
                             mass_smaller_3  mass               0.6538581961218565 - 2.999977935638737   
                             mass_smaller_14 mass              0.6538581961218565 - 283.72102985530483   
                             mass_smaller_30 mass              0.6538581961218565 - 283.72102985530483   

                                                                              RVE  \
model         physical_model filter          output_parameter                       
random_forest MIST           Base            mass              0.9997441818803499   
                             mass_smaller_3  mass              0.9996653648909077   
                             mass_smaller_14 mass               0.999743214302835   
                             mass_smaller_30 mass              0.9997429524189717   

                                                                              RMSE  \
model         physical_model filter          output_parameter                        
random_forest MIST           Base            mass              0.47498420224835197   
                             mass_smaller_3  mass              0.01145266644937612   
                             mass_smaller_14 mass               0.4758809844765971   
                             mass_smaller_30 mass              0.47612846312273854   

                                                                                MAE  \
model         physical_model filter          output_parameter                         
random_forest MIST           Base            mass               0.05965777535357753   
                             mass_smaller_3  mass              0.006816718583654214   
                             mass_smaller_14 mass               0.05943118368961476   
                             mass_smaller_30 mass              0.059361707757584606   

                                                                              MedAE  \
model         physical_model filter          output_parameter                         
random_forest MIST           Base            mass              0.004034495488792533   
                             mass_smaller_3  mass              0.004307804055038389   
                             mass_smaller_14 mass              0.004050206037180315   
                             mass_smaller_30 mass              0.004060477925214023   

                                                                             CORR  \
model         physical_model filter          output_parameter                       
random_forest MIST           Base            mass              0.9998728077084751   
                             mass_smaller_3  mass              0.9998368427876034   
                             mass_smaller_14 mass              0.9998723208690561   
                             mass_smaller_30 mass              0.9998722085378534   

                                                                           MAX_ER  \
model         physical_model filter          output_parameter                       
random_forest MIST           Base            mass              21.388650133880788   
                             mass_smaller_3  mass              0.2152611485477416   
                             mass_smaller_14 mass              23.113277495647964   
                             mass_smaller_30 mass               18.59308616784716   

                                                                                                               Percentiles  \
model         physical_model filter          output_parameter                                                                
random_forest MIST           Base            mass               75 : 0.01064 / 90 : 0.02559 / 95 : 0.07213

Results for the phase category with a value of 3.0


value range  \
model         physical_model filter          output_parameter                                           
random_forest MIST           Base            mass               0.5484495344916944 - 288.814589557117   
                             mass_smaller_3  mass              0.5418086127397849 - 2.999885115629765   
                             mass_smaller_14 mass               0.5484495344916944 - 288.814589557117   
                             mass_smaller_30 mass               0.5484495344916944 - 288.814589557117   

                                                                              RVE  \
model         physical_model filter          output_parameter                       
random_forest MIST           Base            mass              0.9949810881828094   
                             mass_smaller_3  mass              0.9956700597782066   
                             mass_smaller_14 mass              0.9949130093083712   
                             mass_smaller_30 mass              0.9949681865732133   

                                                                              RMSE  \
model         physical_model filter          output_parameter                        
random_forest MIST           Base            mass                1.642420538512268   
                             mass_smaller_3  mass              0.04218948342523731   
                             mass_smaller_14 mass               1.6535157402838208   
                             mass_smaller_30 mass               1.6444983811441365   

                                                                                MAE  \
model         physical_model filter          output_parameter                         
random_forest MIST           Base            mass               0.20145454791412173   
                             mass_smaller_3  mass              0.014501789192269845   
                             mass_smaller_14 mass                0.2013493398604617   
                             mass_smaller_30 mass               0.20111508890417903   

                                                                              MedAE  \
model         physical_model filter          output_parameter                         
random_forest MIST           Base            mass              0.006907232098912752   
                             mass_smaller_3  mass              0.003545841487851087   
                             mass_smaller_14 mass              0.006905876154084667   
                             mass_smaller_30 mass              0.006940862869302067   

                                                                             CORR  \
model         physical_model filter          output_parameter                       
random_forest MIST           Base            mass              0.9976029233414596   
                             mass_smaller_3  mass              0.9978436498641285   
                             mass_smaller_14 mass              0.9975685820026399   
                             mass_smaller_30 mass              0.9975920917628096   

                                                                           MAX_ER  \
model         physical_model filter          output_parameter                       
random_forest MIST           Base            mass               95.14828406903314   
                             mass_smaller_3  mass              1.3490965095666518   
                             mass_smaller_14 mass                99.1425528233774   
                             mass_smaller_30 mass              103.44266579660032   

                                                                                                               Percentiles  \
model         physical_model filter          output_parameter                                                                
random_forest MIST           Base            mass                75 : 0.0288 / 90 : 0.10782 / 95 : 0.29875 / 99

In [4]:
compare_metrics(path, value_rounding=3,
                output_parameters=["mass", "radius"], # output_parameters=["radius"],
                physical_models=["MIST"],
                data_filters=["Base"])

Results for the Global category with a value of 1.0


value range    RVE  \
model             physical_model filter output_parameter                       
decision_tree     MIST           Base   mass                 -298.445  0.995   
                                        radius                 -4.127  1.000   
KNN               MIST           Base   mass                 -298.445  0.990   
                                        radius                 -4.127  1.000   
linear_regression MIST           Base   mass                 -298.445  0.304   
                                        radius                 -4.127  0.985   
mlp               MIST           Base   mass                 -298.445  0.968   
                                        radius                 -4.127  0.989   
random_forest     MIST           Base   mass                 -298.445  0.998   
                                        radius                 -4.127  1.000   
xgboost           MIST           Base   mass                 -298.445  0.994   
                                        radius                 -4.127  1.000   

                                                            RMSE     MAE  \
model             physical_model filter output_parameter                   
decision_tree     MIST           Base   mass               1.854   0.200   
                                        radius             0.011   0.006   
KNN               MIST           Base   mass               2.726   0.328   
                                        radius             0.007   0.004   
linear_regression MIST           Base   mass              22.280  11.345   
                                        radius             0.089   0.065   
mlp               MIST           Base   mass               4.750   1.280   
                                        radius             0.077   0.056   
random_forest     MIST           Base   mass               1.251   0.145   
                                        radius             0.006   0.003   
xgboost           MIST           Base   mass               2.048   0.531   
                                        radius             0.015   0.011   

                                                          MedAE   CORR  \
model             physical_model filter output_parameter                 
decision_tree     MIST           Base   mass              0.003  0.998   
                                        radius            0.003  1.000   
KNN               MIST           Base   mass              0.008  0.995   
                                        radius            0.003  1.000   
linear_regression MIST           Base   mass              8.154  0.552   
                                        radius            0.046  0.993   
mlp               MIST           Base   mass              0.277  0.984   
                                        radius            0.042  0.995   
random_forest     MIST           Base   mass              0.004  0.999   
                                        radius            0.001  1.000   
xgboost           MIST           Base   mass              0.099  0.997   
                                        radius            0.008  1.000   

                                                           MAX_ER  \
model             physical_model filter output_parameter            
decision_tree     MIST           Base   mass              151.658   
                                        radius              0.488   
KNN               MIST           Base   mass              144.214   
                                        radius              0.307   
linear_regression MIST           Base   mass              257.508   
                                        radius              0.525   
mlp               MIST           Base   mass              179.182   
                                        radius              0.692   
random_forest     MIST           Base   mass               95.148   
                                        radius              0.358   
xgboost

Results for the phase category with a value of 0.0


value range    RVE  \
model             physical_model filter output_parameter                       
decision_tree     MIST           Base   mass                 -298.445  0.994   
                                        radius                 -4.080  1.000   
KNN               MIST           Base   mass                 -298.445  0.984   
                                        radius                 -4.080  1.000   
linear_regression MIST           Base   mass                 -298.445  0.273   
                                        radius                 -4.080  0.978   
mlp               MIST           Base   mass                 -298.445  0.965   
                                        radius                 -4.080  0.981   
random_forest     MIST           Base   mass                 -298.445  0.997   
                                        radius                 -4.080  1.000   
xgboost           MIST           Base   mass                 -298.445  0.991   
                                        radius                 -4.080  0.999   

                                                            RMSE     MAE  \
model             physical_model filter output_parameter                   
decision_tree     MIST           Base   mass               2.111   0.270   
                                        radius             0.008   0.005   
KNN               MIST           Base   mass               3.292   0.432   
                                        radius             0.007   0.004   
linear_regression MIST           Base   mass              22.430  13.108   
                                        radius             0.085   0.066   
mlp               MIST           Base   mass               4.923   1.342   
                                        radius             0.079   0.057   
random_forest     MIST           Base   mass               1.352   0.168   
                                        radius             0.004   0.002   
xgboost           MIST           Base   mass               2.494   0.677   
                                        radius             0.014   0.010   

                                                           MedAE   CORR  \
model             physical_model filter output_parameter                  
decision_tree     MIST           Base   mass               0.006  0.997   
                                        radius             0.003  1.000   
KNN               MIST           Base   mass               0.010  0.992   
                                        radius             0.003  1.000   
linear_regression MIST           Base   mass              11.253  0.534   
                                        radius             0.052  0.989   
mlp               MIST           Base   mass               0.290  0.983   
                                        radius             0.043  0.990   
random_forest     MIST           Base   mass               0.004  0.999   
                                        radius             0.001  1.000   
xgboost           MIST           Base   mass               0.126  0.996   
                                        radius             0.007  1.000   

                                                           MAX_ER  \
model             physical_model filter output_parameter            
decision_tree     MIST           Base   mass              131.775   
                                        radius              0.488   
KNN               MIST           Base   mass              126.674   
                                        radius              0.154   
linear_regression MIST           Base   mass              257.508   
                                        radius              0.458   
mlp               MIST           Base   mass              127.141   
                                        radius              0.692   
random_forest     MIST           Base   mass               83.611   
                                        radius              0.

Results for the phase category with a value of 2.0


value range    RVE  \
model             physical_model filter output_parameter                       
decision_tree     MIST           Base   mass                 -283.067  0.999   
                                        radius                 -2.800  1.000   
KNN               MIST           Base   mass                 -283.067  0.999   
                                        radius                 -2.800  1.000   
linear_regression MIST           Base   mass                 -283.067  0.317   
                                        radius                 -2.800  0.968   
mlp               MIST           Base   mass                 -283.067  0.981   
                                        radius                 -2.800  0.981   
random_forest     MIST           Base   mass                 -283.067  1.000   
                                        radius                 -2.800  1.000   
xgboost           MIST           Base   mass                 -283.067  0.999   
                                        radius                 -2.800  0.999   

                                                            RMSE     MAE  \
model             physical_model filter output_parameter                   
decision_tree     MIST           Base   mass               0.856   0.063   
                                        radius             0.010   0.007   
KNN               MIST           Base   mass               0.737   0.095   
                                        radius             0.005   0.003   
linear_regression MIST           Base   mass              24.543  10.236   
                                        radius             0.088   0.058   
mlp               MIST           Base   mass               4.103   1.093   
                                        radius             0.068   0.050   
random_forest     MIST           Base   mass               0.475   0.060   
                                        radius             0.004   0.003   
xgboost           MIST           Base   mass               0.996   0.293   
                                        radius             0.014   0.011   

                                                          MedAE   CORR  \
model             physical_model filter output_parameter                 
decision_tree     MIST           Base   mass              0.000  1.000   
                                        radius            0.005  1.000   
KNN               MIST           Base   mass              0.004  1.000   
                                        radius            0.003  1.000   
linear_regression MIST           Base   mass              5.268  0.596   
                                        radius            0.034  0.984   
mlp               MIST           Base   mass              0.263  0.991   
                                        radius            0.039  0.991   
random_forest     MIST           Base   mass              0.004  1.000   
                                        radius            0.002  1.000   
xgboost           MIST           Base   mass              0.071  0.999   
                                        radius            0.009  1.000   

                                                           MAX_ER  \
model             physical_model filter output_parameter            
decision_tree     MIST           Base   mass              115.074   
                                        radius              0.262   
KNN               MIST           Base   mass               37.487   
                                        radius              0.077   
linear_regression MIST           Base   mass              247.133   
                                        radius              0.453   
mlp               MIST           Base   mass               92.622   
                                        radius              0.403   
random_forest     MIST           Base   mass               21.389   
                                        radius              0.101   
xgboost

Results for the phase category with a value of 3.0


value range    RVE  \
model             physical_model filter output_parameter                       
decision_tree     MIST           Base   mass                 -288.266  0.992   
                                        radius                 -3.474  0.999   
KNN               MIST           Base   mass                 -288.266  0.986   
                                        radius                 -3.474  1.000   
linear_regression MIST           Base   mass                 -288.266  0.388   
                                        radius                 -3.474  0.970   
mlp               MIST           Base   mass                 -288.266  0.952   
                                        radius                 -3.474  0.980   
random_forest     MIST           Base   mass                 -288.266  0.995   
                                        radius                 -3.474  1.000   
xgboost           MIST           Base   mass                 -288.266  0.994   
                                        radius                 -3.474  0.999   

                                                            RMSE    MAE  \
model             physical_model filter output_parameter                  
decision_tree     MIST           Base   mass               2.086  0.199   
                                        radius             0.016  0.008   
KNN               MIST           Base   mass               2.735  0.364   
                                        radius             0.010  0.005   
linear_regression MIST           Base   mass              18.164  7.961   
                                        radius             0.100  0.070   
mlp               MIST           Base   mass               5.083  1.368   
                                        radius             0.082  0.060   
random_forest     MIST           Base   mass               1.642  0.201   
                                        radius             0.009  0.004   
xgboost           MIST           Base   mass               1.730  0.451   
                                        radius             0.019  0.013   

                                                          MedAE   CORR  \
model             physical_model filter output_parameter                 
decision_tree     MIST           Base   mass              0.004  0.996   
                                        radius            0.004  1.000   
KNN               MIST           Base   mass              0.010  0.993   
                                        radius            0.003  1.000   
linear_regression MIST           Base   mass              4.599  0.650   
                                        radius            0.046  0.985   
mlp               MIST           Base   mass              0.264  0.976   
                                        radius            0.046  0.990   
random_forest     MIST           Base   mass              0.007  0.998   
                                        radius            0.002  1.000   
xgboost           MIST           Base   mass              0.094  0.997   
                                        radius            0.009  0.999   

                                                           MAX_ER  \
model             physical_model filter output_parameter            
decision_tree     MIST           Base   mass              151.658   
                                        radius              0.351   
KNN               MIST           Base   mass              144.214   
                                        radius              0.307   
linear_regression MIST           Base   mass              250.918   
                                        radius              0.525   
mlp               MIST           Base   mass              179.182   
                                        radius              0.636   
random_forest     MIST           Base   mass               95.148   
                                        radius              0.292   
xgboost           MIS

## Getting the ranges for the values

In [10]:
iso_handler = Iso_data_handler("C:/Users/antoi/Code/unif/MA2/thesis/data/MIST_v1.2_vvcrit0.0_basic_isos/", 
                                    ['log10_isochrone_age_yr', 'log_Teff', 'log_g', 'phase', 'metallicity', 'star_mass', 'log_R', 'log_L'], 
                                    "MIST", reclassify=True)

iso_df = iso_handler.get_isochrone_dataframe()
display(iso_df)

Reading MIST dataframe from csv file...


,log10_isochrone_age_yr,log_Teff,log_g,phase,metallicity,star_mass,log_R,log_L
0,5.0,3.486221,3.131342,-1.0,-0.25,0.100000,0.153402,-0.793667
1,5.0,3.487362,3.126808,-1.0,-0.25,0.102645,0.160326,-0.775254
2,5.0,3.489243,3.119367,-1.0,-0.25,0.107039,0.171785,-0.744810
3,5.0,3.491102,3.112165,-1.0,-0.25,0.111419,0.183099,-0.714749
4,5.0,3.492937,3.105143,-1.0,-0.25,0.115789,0.194305,-0.684996
...,...,...,...,...,...,...,...,...
1467117,10.3,4.402490,7.777159,6.0,0.50,0.532726,-1.806255,-1.047903
1467118,10.3,4.387132,7.783242,6.0,0.50,0.532730,-1.809295,-1.115415
1467119,10.3,4.371789,7.789130,6.0,0.50,0.532735,-1.812237,-1.182672
1467120,10.3,4.356480,7.794844,6.0,0.50,0.532741,-1.815091,-1.249617


In [17]:
phase_filtered_iso_df = Data_preparator.filter_data(iso_df, {'phase':[0, 2, 3], 'star_mass':("<", 30)})
print_all_uniques(phase_filtered_iso_df)

log10_isochrone_age_yr  : Range : 5.0 - 10.3, Mean : 8.2978, Median : 8.35

log_Teff  : Range : 3.34 - 5.3542, Mean : 3.9845, Median : 3.9191

log_g  : Range : -1.08 - 7.9405, Mean : 3.5437, Median : 3.9117

Values in phase column : 0.0, 2.0, 3.0 

Values in metallicity column : -0.25, -0.5, -0.75, -1.0, -1.25, -1.5, -1.75, -2.0, -2.5, -3.0, -3.5, -4.0, 0.0, 0.25, 0.5 

star_mass  : Range : 0.1 - 29.9991, Mean : 5.0208, Median : 2.4047

log_R  : Range : -1.86 - 3.1298, Mean : 0.6537, Median : 0.6118

log_L  : Range : -3.33 - 5.9763, Mean : 2.2, Median : 2.3306



## Base vs mass_filtered

| Parameter      | Base                                                       | mass_filtered_smaller_3                                | mass_filtered_smaller_14                                           | mass_filtered_smaller_30                               |
| -------------  | ---------------------------------------------------------- | ------------------------------------------------------ | ------------------------------------------------------------------ | ------------------------------------------------------ |
| age            | Range : Range : 5.0 - 10.3 , Mean : 8.1222, Median : 8.15  | Range : 6.2 - 10.3, Mean : 9.1366, Median : 9.2        | Range : 5.0 - 10.3, Mean : 8.4693, Median : 8.5                    | Range : 0.1 - 29.9991, Mean : 5.0208, Median : 2.4047  |
| mass           | Range : 0.1000 - 298.5448, Mean : 11.3677, Median : 2.8697 | Range : 0.1 - 3.0, Mean : 1.3172, Median : 1.1821      | star_mass  : Range : 0.1 - 13.9995, Mean : 3.3831, Median : 2.0289 | Range : 0.1 - 3.0, Mean : 1.4429, Median : 1.3194      |
| log_teff       | Range : 3.34 - 5.3788, Mean : 4.0309, Median : 3.8557      | Range : 3.34 - 5.2138, Mean : 3.8022, Median : 3.7554  | log_Teff  : Range : 3.34 - 5.2138, Mean : 3.9383, Median : 3.867   | Range : 3.34 - 5.3542, Mean : 3.9845, Median : 3.9191  |
| log_g          | Range : -1.08 - 7.9405, Mean : 3.515, Median : 3.8557      | Range : -1.08 - 7.9405, Mean : 3.6564, Median : 4.1712 | log_g  : Range : -1.08 - 7.9405, Mean : 3.5574, Median : 3.9537    | Range : -1.08 - 7.9405, Mean : 3.5437, Median : 3.9117 |
| log_R          | Range : -1.86 - 3.1298, Mean : 0.7296, Median : 0.6976     | Range : -1.86 - 2.7813, Mean : 0.4084, Median : 0.2349 | log_R  : Range : -1.86 - 3.0828, Mean : 0.6007, Median : 0.5357    | Range : -1.86 - 3.1298, Mean : 0.6537, Median : 0.6118 |

### Mass parameter

In [5]:
compare_metrics(path, 
                output_parameters=["mass"],
                physical_models=["MIST"],
                data_filters=["Base", "mass_smaller_3", "mass_smaller_14", "mass_smaller_30"])

Results for the Global category with a value of 1.0


value range  \
model             physical_model filter          output_parameter                                            
decision_tree     MIST           Base            mass               0.0999979840073621 - 298.5447575808816   
                                 mass_smaller_3  mass              0.0999979840073621 - 2.9999868374941867   
                                 mass_smaller_14 mass               0.0999979840073621 - 298.5447575808816   
                                 mass_smaller_30 mass               0.0999979840073621 - 298.5447575808816   
KNN               MIST           Base            mass               0.0999979840073621 - 298.5447575808816   
                                 mass_smaller_3  mass              0.0999979840073621 - 2.9999868374941867   
                                 mass_smaller_14 mass               0.0999979840073621 - 298.5447575808816   
                                 mass_smaller_30 mass               0.0999979840073621 - 298.5447575808816   
linear_regression MIST           Base            mass               0.0999979840073621 - 298.5447575808816   
                                 mass_smaller_3  mass              0.0999979840073621 - 2.9999868374941867   
                                 mass_smaller_14 mass               0.0999979840073621 - 298.5447575808816   
                                 mass_smaller_30 mass               0.0999979840073621 - 298.5447575808816   
mlp               MIST           Base            mass               0.0999979840073621 - 298.5447575808816   
                                 mass_smaller_3  mass              0.0999979840073621 - 2.9999868374941867   
                                 mass_smaller_14 mass               0.0999979840073621 - 298.5447575808816   
                                 mass_smaller_30 mass               0.0999979840073621 - 298.5447575808816   
random_forest     MIST           Base            mass               0.0999979840073621 - 298.5447575808816   
                                 mass_smaller_3  mass              0.0999979840073621 - 2.9999868374941867   
                                 mass_smaller_14 mass               0.0999979840073621 - 298.5447575808816   
                                 mass_smaller_30 mass               0.0999979840073621 - 298.5447575808816   
xgboost           MIST           Base            mass               0.0999979840073621 - 298.5447575808816   
                                 mass_smaller_3  mass              0.0999979840073621 - 2.9999868374941867   
                                 mass_smaller_14 mass               0.0999979840073621 - 298.5447575808816   
                                 mass_smaller_30 mass               0.0999979840073621 - 298.5447575808816   

                                                                                  RVE  \
model             physical_model filter          output_parameter                       
decision_tree     MIST           Base            mass              0.9951817784001245   
                                 mass_smaller_3  mass              0.9986106418886865   
                                 mass_smaller_14 mass              0.9952844938510328   
                                 mass_smaller_30 mass              0.9951639894597079   
KNN               MIST           Base            mass              0.9895916504979603   
                                 mass_smaller_3  mass              0.9984125750212497   
                                 mass_smaller_14 mass              0.9895916504979603   
                                 mass_smaller_30 mass              0.9895916504979603   
linear_regression MIST           Base            mass              0.3042271481418729   
                                 mass_smaller_3  mass              0.6829393130134978   
                                 mass_smaller_14 mass              0.3042271481418729   
                                 mass_smaller_30 mass              0.30422714

Results for the phase category with a value of 0.0


value range  \
model             physical_model filter          output_parameter                                            
decision_tree     MIST           Base            mass               0.0999979840073621 - 298.5447575808816   
                                 mass_smaller_3  mass              0.0999979840073621 - 2.9999868374941867   
                                 mass_smaller_14 mass               0.0999979840073621 - 298.5447575808816   
                                 mass_smaller_30 mass               0.0999979840073621 - 298.5447575808816   
KNN               MIST           Base            mass               0.0999979840073621 - 298.5447575808816   
                                 mass_smaller_3  mass              0.0999979840073621 - 2.9999868374941867   
                                 mass_smaller_14 mass               0.0999979840073621 - 298.5447575808816   
                                 mass_smaller_30 mass               0.0999979840073621 - 298.5447575808816   
linear_regression MIST           Base            mass               0.0999979840073621 - 298.5447575808816   
                                 mass_smaller_3  mass              0.0999979840073621 - 2.9999868374941867   
                                 mass_smaller_14 mass               0.0999979840073621 - 298.5447575808816   
                                 mass_smaller_30 mass               0.0999979840073621 - 298.5447575808816   
mlp               MIST           Base            mass               0.0999979840073621 - 298.5447575808816   
                                 mass_smaller_3  mass              0.0999979840073621 - 2.9999868374941867   
                                 mass_smaller_14 mass               0.0999979840073621 - 298.5447575808816   
                                 mass_smaller_30 mass               0.0999979840073621 - 298.5447575808816   
random_forest     MIST           Base            mass               0.0999979840073621 - 298.5447575808816   
                                 mass_smaller_3  mass              0.0999979840073621 - 2.9999868374941867   
                                 mass_smaller_14 mass               0.0999979840073621 - 298.5447575808816   
                                 mass_smaller_30 mass               0.0999979840073621 - 298.5447575808816   
xgboost           MIST           Base            mass               0.0999979840073621 - 298.5447575808816   
                                 mass_smaller_3  mass              0.0999979840073621 - 2.9999868374941867   
                                 mass_smaller_14 mass               0.0999979840073621 - 298.5447575808816   
                                 mass_smaller_30 mass               0.0999979840073621 - 298.5447575808816   

                                                                                  RVE  \
model             physical_model filter          output_parameter                       
decision_tree     MIST           Base            mass              0.9935567994940296   
                                 mass_smaller_3  mass              0.9998109603044399   
                                 mass_smaller_14 mass              0.9936483630983785   
                                 mass_smaller_30 mass              0.9935794089969434   
KNN               MIST           Base            mass              0.9843387655163102   
                                 mass_smaller_3  mass              0.9992443930168096   
                                 mass_smaller_14 mass              0.9843387655163102   
                                 mass_smaller_30 mass              0.9843387655163102   
linear_regression MIST           Base            mass              0.2728709030098654   
                                 mass_smaller_3  mass              0.8226816036408051   
                                 mass_smaller_14 mass              0.2728709030098654   
                                 mass_smaller_30 mass              0.27287090

Results for the phase category with a value of 2.0


value range  \
model             physical_model filter          output_parameter                                            
decision_tree     MIST           Base            mass              0.6538581961218565 - 283.72102985530483   
                                 mass_smaller_3  mass               0.6538581961218565 - 2.999977935638737   
                                 mass_smaller_14 mass              0.6538581961218565 - 283.72102985530483   
                                 mass_smaller_30 mass              0.6538581961218565 - 283.72102985530483   
KNN               MIST           Base            mass              0.6538581961218565 - 283.72102985530483   
                                 mass_smaller_3  mass               0.6538581961218565 - 2.999977935638737   
                                 mass_smaller_14 mass              0.6538581961218565 - 283.72102985530483   
                                 mass_smaller_30 mass              0.6538581961218565 - 283.72102985530483   
linear_regression MIST           Base            mass              0.6538581961218565 - 283.72102985530483   
                                 mass_smaller_3  mass               0.6538581961218565 - 2.999977935638737   
                                 mass_smaller_14 mass              0.6538581961218565 - 283.72102985530483   
                                 mass_smaller_30 mass              0.6538581961218565 - 283.72102985530483   
mlp               MIST           Base            mass              0.6538581961218565 - 283.72102985530483   
                                 mass_smaller_3  mass               0.6538581961218565 - 2.999977935638737   
                                 mass_smaller_14 mass              0.6538581961218565 - 283.72102985530483   
                                 mass_smaller_30 mass              0.6538581961218565 - 283.72102985530483   
random_forest     MIST           Base            mass              0.6538581961218565 - 283.72102985530483   
                                 mass_smaller_3  mass               0.6538581961218565 - 2.999977935638737   
                                 mass_smaller_14 mass              0.6538581961218565 - 283.72102985530483   
                                 mass_smaller_30 mass              0.6538581961218565 - 283.72102985530483   
xgboost           MIST           Base            mass              0.6538581961218565 - 283.72102985530483   
                                 mass_smaller_3  mass               0.6538581961218565 - 2.999977935638737   
                                 mass_smaller_14 mass              0.6538581961218565 - 283.72102985530483   
                                 mass_smaller_30 mass              0.6538581961218565 - 283.72102985530483   

                                                                                  RVE  \
model             physical_model filter          output_parameter                       
decision_tree     MIST           Base            mass              0.9991682912199935   
                                 mass_smaller_3  mass              0.9992078967946935   
                                 mass_smaller_14 mass              0.9992190516533151   
                                 mass_smaller_30 mass              0.9992529638955677   
KNN               MIST           Base            mass              0.9993836364150782   
                                 mass_smaller_3  mass              0.9991876883590323   
                                 mass_smaller_14 mass              0.9993836364150782   
                                 mass_smaller_30 mass              0.9993836364150782   
linear_regression MIST           Base            mass              0.3168817813892185   
                                 mass_smaller_3  mass              0.6094803842541214   
                                 mass_smaller_14 mass              0.3168817813892185   
                                 mass_smaller_30 mass              0.31688178

Results for the phase category with a value of 3.0


value range  \
model             physical_model filter          output_parameter                                           
decision_tree     MIST           Base            mass               0.5484495344916944 - 288.814589557117   
                                 mass_smaller_3  mass              0.5418086127397849 - 2.999885115629765   
                                 mass_smaller_14 mass               0.5484495344916944 - 288.814589557117   
                                 mass_smaller_30 mass               0.5484495344916944 - 288.814589557117   
KNN               MIST           Base            mass               0.5484495344916944 - 288.814589557117   
                                 mass_smaller_3  mass              0.5418086127397849 - 2.999885115629765   
                                 mass_smaller_14 mass               0.5484495344916944 - 288.814589557117   
                                 mass_smaller_30 mass               0.5484495344916944 - 288.814589557117   
linear_regression MIST           Base            mass               0.5484495344916944 - 288.814589557117   
                                 mass_smaller_3  mass              0.5418086127397849 - 2.999885115629765   
                                 mass_smaller_14 mass               0.5484495344916944 - 288.814589557117   
                                 mass_smaller_30 mass               0.5484495344916944 - 288.814589557117   
mlp               MIST           Base            mass               0.5484495344916944 - 288.814589557117   
                                 mass_smaller_3  mass              0.5418086127397849 - 2.999885115629765   
                                 mass_smaller_14 mass               0.5484495344916944 - 288.814589557117   
                                 mass_smaller_30 mass               0.5484495344916944 - 288.814589557117   
random_forest     MIST           Base            mass               0.5484495344916944 - 288.814589557117   
                                 mass_smaller_3  mass              0.5418086127397849 - 2.999885115629765   
                                 mass_smaller_14 mass               0.5484495344916944 - 288.814589557117   
                                 mass_smaller_30 mass               0.5484495344916944 - 288.814589557117   
xgboost           MIST           Base            mass               0.5484495344916944 - 288.814589557117   
                                 mass_smaller_3  mass              0.5418086127397849 - 2.999885115629765   
                                 mass_smaller_14 mass               0.5484495344916944 - 288.814589557117   
                                 mass_smaller_30 mass               0.5484495344916944 - 288.814589557117   

                                                                                   RVE  \
model             physical_model filter          output_parameter                        
decision_tree     MIST           Base            mass               0.9918974335576843   
                                 mass_smaller_3  mass               0.9925088328108644   
                                 mass_smaller_14 mass               0.9921593929956303   
                                 mass_smaller_30 mass               0.9914993723370994   
KNN               MIST           Base            mass               0.9860770182495742   
                                 mass_smaller_3  mass                 0.99350766733889   
                                 mass_smaller_14 mass               0.9860770182495742   
                                 mass_smaller_30 mass               0.9860770182495742   
linear_regression MIST           Base            mass              0.38830292902171926   
                                 mass_smaller_3  mass              0.15358248802528074   
                                 mass_smaller_14 mass              0.38830292902171926   
                                 mass_smaller_30 mass              0.38830292902171926   

Linear regression : 
- mass_filtered_smaller_30 : similar results as Base (proportionnaly to the mass)
- mass_filtered_smaller_3 : similar results as Base (proportionnaly to the mass)

Decision tree : 
- mass_filtered_smaller_30 : worse results than Base (proportionnaly to the mass)
- mass_filtered_smaller_3 : worse results than Base (proportionnaly to the mass)

KNN : 
- mass_filtered_smaller_30 : slightly better results than Base (proportionnaly to the mass)
- mass_filtered_smaller_3 : slightly worse results than Base (proportionnaly to the mass)

Random forest : 
- mass_filtered_smaller_30 : similar results as Base (proportionnaly to the mass)
- mass_filtered_smaller_3 : slightly worse results than Base (proportionnaly to the mass)

XGBoost : 
- mass_filtered_smaller_30 : slightly worse results than Base (proportionnaly to the mass)
- mass_filtered_smaller_3 : slightly worse results than Base (proportionnaly to the mass)

MLP : 
- mass_filtered_smaller_30 : similar results as Base (proportionnaly to the mass)
- mass_filtered_smaller_3 : similar results as Base (proportionnaly to the mass)

Linear regression : 
- mass_smaller_3 : 
- mass_smaller_14 : 
- mass_smaller_30 : 

Decision tree : 
- mass_smaller_3 : 
- mass_smaller_14 : 
- mass_smaller_30 : 

KNN : 
- mass_smaller_3 : 
- mass_smaller_14 : 
- mass_smaller_30 : 

Random forest : 
- mass_smaller_3 : 
- mass_smaller_14 : 
- mass_smaller_30 : 

XGBoost : 
- mass_smaller_3 : 
- mass_smaller_14 : 
- mass_smaller_30 : 

MLP : 
- mass_smaller_3 : 
- mass_smaller_14 : 
- mass_smaller_30 : 

#### Conclusions on mass

Proportionnaly to the mass, both filtering give similar (sometimes slightly worse or better) results than Base

&emsp; => not interesting to use as it also restricts what kind of stars the model can predict

### Radius parameter

In [ ]:
compare_metrics(path, 
                output_parameters=["radius"],
                physical_models=["MIST"],
                data_filters=["Base", "mass_filtered_smaller_30", "mass_filtered_smaller_3"])

RVE  \
model             physical_model filter                   output_parameter                       
decision_tree     MIST           Base                     radius            0.9997899702074513   
                                 mass_filtered_smaller_30 radius            0.9998061270554189   
                                 mass_filtered_smaller_3  radius            0.9997980449596693   
KNN               MIST           Base                     radius             0.999833901666557   
                                 mass_filtered_smaller_30 radius             0.999845742270286   
                                 mass_filtered_smaller_3  radius            0.9998124002057707   
linear_regression MIST           Base                     radius            0.9897611137480969   
                                 mass_filtered_smaller_30 radius            0.9925428351690286   
                                 mass_filtered_smaller_3  radius            0.9958055391259805   
mlp               MIST           Base                     radius            0.9960419184441298   
                                 mass_filtered_smaller_30 radius            0.9987759834874893   
                                 mass_filtered_smaller_3  radius            0.9994078667866224   
random_forest     MIST           Base                     radius             0.999904453912182   
                                 mass_filtered_smaller_30 radius             0.999911027099752   
                                 mass_filtered_smaller_3  radius            0.9998952156558059   
xgboost           MIST           Base                     radius            0.9996009897646162   
                                 mass_filtered_smaller_30 radius             0.999676980016293   
                                 mass_filtered_smaller_3  radius            0.9997374331207423   

                                                                                            RMSE  \
model             physical_model filter                   output_parameter                         
decision_tree     MIST           Base                     radius            0.013774959488591745   
                                 mass_filtered_smaller_30 radius            0.013462255965888063   
                                 mass_filtered_smaller_3  radius            0.014543308792759283   
KNN               MIST           Base                     radius            0.012250269140419087   
                                 mass_filtered_smaller_30 radius               0.012009180907773   
                                 mass_filtered_smaller_3  radius            0.014019621430573385   
linear_regression MIST           Base                     radius             0.09617751115675367   
                                 mass_filtered_smaller_30 radius              0.0834915224254678   
                                 mass_filtered_smaller_3  radius               0.066277386089521   
mlp               MIST           Base                     radius            0.060330025886507314   
                                 mass_filtered_smaller_30 radius            0.034357031439120785   
                                 mass_filtered_smaller_3  radius             0.02789966474335524   
random_forest     MIST           Base                     radius            0.009291129066072832   
                                 mass_filtered_smaller_30 radius            0.009120128374631557   
                                 mass_filtered_smaller_3  radius            0.010476065091138284   
xgboost           MIST           Base                     radius            0.018986255054897025   
                                 mass_filtered_smaller_30 radius             0.01737682271670121   
                                 mass_filtered_smaller_3  radius             0.01658239941408444   

                                                                                              MAE  \
model             physical_m

Linear regression : 
- mass_filtered_smaller_30 : slightly better results than Base
- mass_filtered_smaller_3 : slightly better results than Base

Decision tree : 
- mass_filtered_smaller_30 : similar results as Base but worse MAX_ER
- mass_filtered_smaller_3 : slightly better results than Base

KNN : 
- mass_filtered_smaller_30 : slightly better results than Base
- mass_filtered_smaller_3 : similar results as Base

Random forest : 
- mass_filtered_smaller_30 : slightly better results than Base
- mass_filtered_smaller_3 : similar results as Base

XGBoost : 
- mass_filtered_smaller_30 : slightly better results than Base
- mass_filtered_smaller_3 : better results than Base

MLP : 
- mass_filtered_smaller_30 : better results than Base
- mass_filtered_smaller_3 : better results than Base

Linear regression : 
- mass_smaller_3 : 
- mass_smaller_14 : 
- mass_smaller_30 : 

Decision tree : 
- mass_smaller_3 : 
- mass_smaller_14 : 
- mass_smaller_30 : 

KNN : 
- mass_smaller_3 : 
- mass_smaller_14 : 
- mass_smaller_30 : 

Random forest : 
- mass_smaller_3 : 
- mass_smaller_14 : 
- mass_smaller_30 : 

XGBoost : 
- mass_smaller_3 : 
- mass_smaller_14 : 
- mass_smaller_30 : 

MLP : 
- mass_smaller_3 : 
- mass_smaller_14 : 
- mass_smaller_30 : 

#### Conclusions on radius

The results of both filtering yielded are similar to Base.

&emsp; => not useful to use these filtering instead of Base

### Conclusion on both

Both parameter give similar results as Base while having the restriction on the mass that the model will be able to predict.

&emsp; => It is not useful to use this filtering instead of Base

## Final conclusion on the filters

Filtering on the log_g => not useful as the results of the two parameters go in opposite directions (when one improves, the other worsens)

Filtering on the age => age_filtered_bigger_7 improves the results of both parameters compared to Base and might be useful to use if we are fine with losing the possibility to predict for stars below 10.000.000 years

Filtering on the mass => not useful to use this filtering as it gives similar results as Base while being restricted in the kind of stars it can predict

Filtering on the phase => not useful to use this filtering as it gives similar results as Base while being restricted in the kind of stars it can predict

Filtering on the radius => not useful to use this filtering as it gives worse results than Base overall

Filtering on the Teff => not useful as the results of the two parameters go in opposite directions (when one improves, the other worsens)